storybookjs/storybook
# Allow passing options to react-test-renderer .create() method

Issue:
When running storyshots with a story for a component that uses refs, it will throw a null reference error (see https://github.com/facebook/react/issues/7371)

## What I did
This PR adds the ability to pass options to the react-test-renderer create method, for example to allow mocking refs as per https://facebook.github.io/react/blog/2016/11/16/react-v15.4.0.html#mocking-refs-for-snapshot-testing


## How to test
Run `yarn test` - `ComponentWithRef.stories.js` should use the `createNodeMock` function defined in `storyshots.test.js` to mock its ref and not fail with a null reference error. The story for `ComponentWithRef` attempts to get the scrollWidth of a div when the component mounts.
